File: src/main/java/com/github/reviversmc/modget/minecraft/config/ModMenuConfig.java
package com.github.reviversmc.modget.minecraft.config;

import com.terraformersmc.modmenu.api.ConfigScreenFactory;
import com.terraformersmc.modmenu.api.ModMenuApi;
import me.shedaniel.clothconfig2.api.ConfigBuilder;
import me.shedaniel.clothconfig2.api.ConfigCategory;
import me.shedaniel.clothconfig2.api.ConfigEntryBuilder;
import net.minecraft.text.LiteralText;

import java.io.IOException;

public class ModMenuConfig implements ModMenuApi {
    @Override
    public ConfigScreenFactory<?> getModConfigScreenFactory() {
        return parent -> {
            ConfigBuilder builder = ConfigBuilder.create()
                .setParentScreen(parent)
                .setTitle(new LiteralText("Auto Check Updates"));

            ConfigCategory general = builder.getOrCreateCategory(new LiteralText("General"));
            builder.setFallbackCategory(general);

            ConfigEntryBuilder entryBuilder = builder.entryBuilder();
            general.addEntry(
                entryBuilder.startBooleanToggle(new LiteralText("Auto Check Updates"), ConfigManager.INSTANCE.getBooleanProperty("autoCheck"))
                .setSaveConsumer(b -> {
                    try {
                        ConfigManager.INSTANCE.setValue("autoCheck", b.toString());
                    } catch (IOException e) {
                        e.printStackTrace();
                    }
                })
                .build());
            return builder.build();
        };
    }
}

File: src/main/java/com/github/reviversmc/modget/minecraft/config/ModMenuConfig.kt
//package com.github.pozitp.config
//
//import com.terraformersmc.modmenu.api.ConfigScreenFactory
//import com.terraformersmc.modmenu.api.ModMenuApi
//import me.shedaniel.clothconfig2.api.ConfigBuilder
//import net.minecraft.client.gui.screen.Screen
//import net.minecraft.text.TranslatableText
//
//
//class ModMenuConfig : ModMenuApi {
//    override fun getModConfigScreenFactory(): ConfigScreenFactory<*> {
//        return ConfigScreenFactory { parent: Screen? ->
//            val builder = ConfigBuilder.create()
//                .setParentScreen(parent)
//                .setTitle(TranslatableText("title.fabrilousupdater.config"))
//            val general = builder.getOrCreateCategory(TranslatableText("category.fabrilousupdater.general"))
//            builder.setFallbackCategory(general)
//
//            val entryBuilder = builder.entryBuilder()
//
//            general.addEntry(
//                    entryBuilder.startBooleanToggle(TranslatableText("option.fabrilousupdater.autoCheck"), ConfigManager.INSTANCE.getBooleanProperty("autoCheck"))
//                            .setSaveConsumer {b: Boolean -> ConfigManager.INSTANCE.setValue("autoCheck", b.toString())}
//                    .build())
//
//
//            builder.build()
//        }
//    }
//}
